### PR TITLE
Customized default setting

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/longhorn/longhorn/issues/623

Design doc: https://docs.google.com/document/d/1eBd5_W-rOrlh66GcIJd5Dl0H0kxSsxG9hhHKJB-jhgc/edit?usp=sharing

### CODE REVIEW

File: app/daemon.go [hunk 1]
```diff
@@ -78,12 +86,18 @@ func startManager(c *cli.Context) error {
...
 
+	defaultSettingPath := c.String(FlagDefaultSettingPath)
+
 	if err := environmentCheck(); err != nil {
 		logrus.Errorf("Failed environment check, please make sure you " +
 			"have iscsiadm/open-iscsi installed on the host")
 		return fmt.Errorf("Environment check failed: %v", err)
 	}
 
+	if _, err := os.Stat(defaultSettingPath); err != nil {
```
Comment (yasker) [inline app/daemon.go]: It should be allowed to be empty so we just use built-in internal defaults.

Comment (yasker) [reply app/daemon.go]: I meant, if the path is empty, we should allow it.

But we indeed should error out if it's specified but cannot be found.

File: datastore/longhorn.go [hunk 2]
```diff
@@ -38,7 +40,19 @@ var (
...
 
-func (s *DataStore) InitSettings() error {
+func (s *DataStore) InitSettings(settingPath string) error {
+	defaultSettingMap := map[string]string{}
+	if settingPath != "" {
+		data, err := ioutil.ReadFile(settingPath)
+		if err != nil {
+			logrus.Errorf("Failed to read default setting file %v, will use built-in internal default settings: %v", settingPath, err)
+		} else {
+			if err := yaml.Unmarshal(data, &defaultSettingMap); err != nil {
```
Comment (yasker) [inline datastore/longhorn.go]: From the doc, it seems yaml.Unmarshal can result in a partial result. We shouldn't allow it.

Comment (shuo-wu) [reply datastore/longhorn.go]: If the code fails here, should we continue to initialize the settings?

Comment (yasker) [reply datastore/longhorn.go]: If we cannot find the file, we should fail since it's a user error.

**Pull request merged**